packages/formik/src/Formik.tsx

```diff
@@ -1020,14 +1020,22 @@ export function useFormik<Values extends FormikValues = FormikValues>({
     validateOnMount,
   };
 
-  return ctx;
+  // We want useFormik to always return the same object reference
+  // so that it can be used as a dependency for other hooks.
+  const ctxRef = React.useRef<typeof ctx>({} as any);
+  Object.assign(ctxRef.current, ctx);
```

Comment:
You can't write to refs in render in concurrent mode. So this strategy won't work in future versions of React

Reply:
Thank, @jeffbski. I don't know its=) I found a good article about it if someone who will read this comment is interesting https://frontarm.com/daishi-kato/use-ref-in-concurrent-mode/#:~:text=Concurrent%20Mode%20lets%20React%20apps,%2C%20text%20input%20or%20hover).

Reply:
But it seems like how exactly in this case the competitive mode will not break due to the use of refs, since several renders will undermine the return of similar objects. But, I agree, we need to fix it =)

Reply:
As far as I understand, there will be a problem with concurrent mode if the reference changes to a different value during render. But in our case, no matter how many times the render is launched, the link will contain a similar object. Plus, before, between renders, a new object was returned every time.

There is an article about it https://frontarm.com/daishi-kato/use-ref-in-concurrent-mode/#:~:text=Concurrent%20Mode%20lets%20React%20apps,%2C%20text%20input% 20or% 20hover). And it clearly shows that the problem arises due to the fact that between renders there are changes in the reference to a new value. There is no such thing. But maybe I don't understand something =))